starter_code/src/movies.js

```diff
@@ -1,12 +1,41 @@
 /* eslint no-restricted-globals: 'off' */
 // Turn duration of the movies from hours to minutes 
-
+var durationInLetters = movies.map(function(movies){
+  return movies.duration.split('');
+});
+var horasSinLetras= durationInLetters.map(function(elem, i){
+  return (durationInLetters[i][0] * 60);
+});
+var minutosSinLetras= durationInLetters.map(function(elem,i){
+  if (elem.indexOf("m")!=-1){
+      return elem.splice(elem.indexOf(' ')+1,elem.indexOf('m')-elem.indexOf(' ')-1).join("");
+  } else {
+   return 0;
+  }
+});
+var tiempoEnMinutos = horasSinLetras.map(function(elem, i){
+  return horasSinLetras[i] + parseInt(minutosSinLetras[i]);
+});
+movies.map(function(elem,i){
+  return elem.duration === tiempoEnMinutos[i];
+});
 
 // Get the average of all rates with 2 decimals 
-
+var ratesAverage = movies.reduce(function(acumulador, current){
+  return acumulador + current.rate / movies.length;
+}, 0);
+ratesAverage = Math.round(ratesAverage * 100) / 100;
+console.log(ratesAverage);
 
 // Get the average of Drama Movies
-
+var dramaMovies = movies.filter(function(elem){
+  return elem.genre.includes('Drama');
+  });
+  var dramaMoviesRate = dramaMovies.reduce(function(acc,cur){
+  return acc + cur.rate/dramaMovies.length;
+  },0);
+  dramaMoviesRate = Math.round(dramaMoviesRate * 100) / 100;
+  console.log(dramaMoviesRate);
 
```

Comment:
Nuevamente, la lógica está bien, pero sería mucho mejor usar funciones. En esta iteración en particular, podían utilizar a la función de la iteración 2 que te da el promedio de un array, para no repetir código.
Les muestro una posible solución utilizando funciones:

// Iteration 2: Get the average of all rates with 2 decimals 
function **ratesAverage(array)**{
    var average = array.reduce(function(sum, movie){
      return sum += parseFloat(movie.rate);
    }, 0);
    return average/array.length;
} 

// Iteration : Get the average of Drama Movies
function dramaMoviesRate(array){
    var drama = array.filter(function(movie){
      for(var i = 0; i < movie.genre.length; i++){
      return movie.genre[i] === "Drama";
      }
    });
    return **ratesAverage(drama)**;
  }